### PR TITLE
[20497] Add related versions table

### PR DESCRIPTION
**Pull request by JesusPoderoso**
This PR is a proposal to include a table with dependencies and other related software product versions in a new section of the documentation.

### CODE REVIEW

**Review by EduPonz — CHANGES_REQUESTED** [review]
I agree with this initiative, but I think we ought to do it some place else so it has more visibility.

File: docs/notes/versions.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.. _related_versions:
+
+Related versions
```
Comment (elianalf) [inline docs/notes/versions.rst:3]: Maybe we can call it Related product versions or Related dependency versions, what do you think?

File: docs/notes/versions.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Fast DDS is continuously evolving and improving.
+This means that the different software products that are part of the Fast DDS ecosystem are evolving and improving
+together with Fast DDS.
+This section provides information about the versions of the eProsima software products related to the Fast DDS.
+
+Build time dependencies
+-----------------------
+
+The following table shows the the different versions of the Fast DDS software dependencies.
```
Comment (elianalf) [inline docs/notes/versions.rst:14]:
```suggestion
The following table shows the corresponding versions of the Fast DDS dependencies.
```

File: docs/notes/notes.rst [hunk 3]
```diff
@@ -30,6 +30,10 @@ This patch release includes the following **fixes**:
   When upgrading to version 2.13.2 it is **advisable** to regenerate generated source from IDL files
    using `Fast DDS-Gen v3.2.1 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v3.2.1>`_.
 
+.. seealso::
+  For further information about this version dependencies and other products related versions, please check the Fast
+  DDS :ref:`related_versions` section.
```
Comment (elianalf) [inline docs/notes/notes.rst:35]: If the suggestion  about the name of the `related_versions` section is applied, please change also here. Moreover, if we change the name we can refer directly to the specific section inside the page, i.e. `Build time dependencies`

File: docs/installation/sources/sources_windows.rst [hunk 4]
```diff
@@ -124,6 +124,10 @@ Dependencies
 * :ref:`asiotinyxml2_sw`
 * :ref:`openssl_sw`
 
+.. seealso::
+  For further information about this Fast DDS version dependencies and other products related versions, please check
+  the Fast DDS :ref:`related_versions` section.
```
Comment (elianalf) [inline docs/installation/sources/sources_windows.rst:129]: If the suggestion  about the name of the `related_versions` section is applied, please change also here. Moreover, if we change the name we can refer directly to the specific section inside the page, i.e. `Build time dependencies`

File: docs/installation/sources/sources_mac.rst [hunk 5]
```diff
@@ -118,6 +118,10 @@ Dependencies
 * :ref:`asiotinyxml2_sm`
 * :ref:`openssl_sm`
 
+.. seealso::
+  For further information about this Fast DDS version dependencies and other products related versions, please check
+  the Fast DDS :ref:`related_versions` section.
```
Comment (elianalf) [inline docs/installation/sources/sources_mac.rst:123]: If the suggestion  about the name of the `related_versions` section is applied, please change also here. Moreover, if we change the name we can refer directly to the specific section inside the page, i.e. `Build time dependencies`

File: docs/installation/sources/sources_linux.rst [hunk 6]
```diff
@@ -69,6 +69,10 @@ Dependencies
 * :ref:`gtest_sl` [optional]
 * :ref:`pythonreq_sl` [optional]
 
+.. seealso::
+  For further information about this Fast DDS version dependencies and other products related versions, please check
+  the Fast DDS :ref:`related_versions` section.
```
Comment (elianalf) [inline docs/installation/sources/sources_linux.rst:74]: If the suggestion  about the name of the `related_versions` section is applied, please change also here. Moreover, if we change the name we can refer directly to the specific section inside the page, i.e. `Build time dependencies`

File: docs/notes/versions.rst [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+Related versions
+================
+
+Fast DDS is continuously evolving and improving.
+This means that the different software products that are part of the Fast DDS ecosystem are evolving and improving
+together with Fast DDS.
+This section provides information about the versions of the eProsima software products related to the Fast DDS.
+
+Build time dependencies
```
Comment (elianalf) [inline docs/notes/versions.rst:11]: Dependecy versions ?

**Review by elianalf — COMMENTED** [review]
I only left few suggestions, changes are not required. It would be nice if we added a "See also section" with a link to the related versions also in the [section for the installation of FastDDSGen](https://eprosima-fast-rtps--652.org.readthedocs.build/en/652/installation/sources/sources_linux.html#fast-dds-gen-installation) (like the one added in installation of dependencies section)

File: docs/notes/versions.rst (hunk 1 above)
Comment (JesusPoderoso) [reply docs/notes/versions.rst:3]: As it applies (and will apply) both dependencies (build time and run time), and other repos compatibility, I think just "Related versions" may suit better. What do you think @EduPonz?

Comment (JesusPoderoso) [reply docs/notes/versions.rst:3]: I mean, this title is relevant and should be short because is the one displayed in the left section/menu

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

File: docs/02-formalia/titlepage.rst [hunk 8]
```diff
@@ -168,6 +168,23 @@ Main Features
...
 
+Dependencies and compatibilities
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*eProsima Fast DDS* is continuously evolving and improving.
+This means that the different software products that are part of the Fast DDS ecosystem are evolving and improving
+together with Fast DDS.
+
+Fast DDS has some :ref:`library dependencies <dependencies_compatibilities_library_dependencies>` as the previously
+mentioned Fast CDR for data serialization, or OpenSSL for secure communication, for instance.
```
Comment (EduPonz) [inline docs/02-formalia/titlepage.rst:179]:
```suggestion
Fast DDS has some :ref:`library dependencies <dependencies_compatibilities_library_dependencies>`, e.g. the previously
mentioned Fast CDR for data serialization, or OpenSSL for secure communications.
```

File: docs/02-formalia/titlepage.rst [hunk 9]
```diff
@@ -168,6 +168,23 @@ Main Features
...
+Basing on different :ref:`platform support levels <dependencies_compatibilities_platform_support>`, it has also
```
Comment (EduPonz) [inline docs/02-formalia/titlepage.rst:180]:
```suggestion
Depending on different :ref:`platform support levels <dependencies_compatibilities_platform_support>`, it has also
```

File: docs/02-formalia/titlepage.rst [hunk 10]
```diff
@@ -168,6 +168,23 @@ Main Features
...
+different :ref:`build dependencies <dependencies_compatibilities_build_system_dependencies>`.
+
+Finally, there are some other *eProsima* products that are use Fast DDS as a middleware, such as *Micro XRCE-DDS*,
```
Comment (EduPonz) [inline docs/02-formalia/titlepage.rst:183]:
```suggestion
Finally, there are some other *eProsima* products that use Fast DDS as a middleware, such as *Micro XRCE-DDS*,
```

File: docs/02-formalia/titlepage.rst [hunk 11]
```diff
@@ -168,6 +168,23 @@ Main Features
...
+*DDS Router* and *Fast DDS python* wrapper.
+Those which are strongly attached to each Fast DDS supported version are described in :ref:`this product compatibility
```
Comment (EduPonz) [inline docs/02-formalia/titlepage.rst:185]:
```suggestion
Those that are strongly attached to each Fast DDS supported version are described in :ref:`this product compatibility
```

File: docs/fastdds/getting_started/getting_started.rst [hunk 12]
```diff
@@ -13,3 +13,8 @@ application.
     definitions
     simple_app/simple_app
     simple_python_app/simple_python_app
+
+.. seealso::
+
+    Check Fast DDS dependencies, libraries and related packages in the :ref:`dependencies_compatibilities` section.
```
Comment (EduPonz) [inline docs/fastdds/getting_started/getting_started.rst:19]:
```suggestion
    Check Fast DDS dependencies, libraries, and related packages in the :ref:`dependencies_compatibilities` section.
```

File: docs/installation/sources/sources_linux.rst [hunk 13]
```diff
@@ -603,6 +608,11 @@ Gradle is an open-source build automation tool which requires a Java version to
     *Fast DDS-Gen* introduced support for Gradle 7 in release v2.2.0.
     Gradle 8 is not yet supported.
 
+.. seealso::
+
+    For further information about Fast DDS-Gen products related versions, please check the
```
Comment (EduPonz) [inline docs/installation/sources/sources_linux.rst:613]:
```suggestion
    For further information about Fast DDS-Gen product related versions, please refer to the
```

File: docs/installation/sources/sources_linux.rst [hunk 14]
```diff
@@ -69,6 +69,11 @@ Dependencies
 * :ref:`gtest_sl` [optional]
 * :ref:`pythonreq_sl` [optional]
 
+.. seealso::
+
+    For further information about this Fast DDS version dependencies and other products related versions, please check
```
Comment (EduPonz) [inline docs/installation/sources/sources_linux.rst:74]: Applies to all sources instructions
```suggestion
    For further information about this Fast DDS version dependencies, as well as for the corresponding versions of other related products, please refer to
```

File: docs/notes/notes.rst [hunk 15]
```diff
@@ -27,8 +27,14 @@ This patch release includes the following **fixes**:
...
 .. note::
-  When upgrading to version 2.13.3 it is **advisable** to regenerate generated source from IDL files
-  using `Fast DDS-Gen v3.2.1 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v3.2.1>`_.
+
+    When upgrading to version 2.13.3 it is **advisable** to regenerate generated source from IDL files
+    using `Fast DDS-Gen v3.2.1 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v3.2.1>`_.
+
+.. seealso::
+
+    For further information about other products versions related to this version, please check the
```
Comment (EduPonz) [inline docs/notes/notes.rst:36]: Apply everywhere
```suggestion
    For further information about the corresponding versions of other products this Fast DDS version, please refer to the
```

File: docs/notes/versions.rst [hunk 16]
```diff
@@ -0,0 +1,388 @@
+.. _dependencies_compatibilities:
+
+Dependencies and compatibilities
+================================
+
+Fast DDS is continuously evolving and improving.
+This means that the different software products that are part of the Fast DDS ecosystem are evolving and improving
+together with Fast DDS.
+This section provides information about the required dependencies for building Fast DDS, and versions of the eProsima
+software products related to the Fast DDS.
```
Comment (EduPonz) [inline docs/notes/versions.rst:10]:
```suggestion
This section provides information about the required dependencies for building Fast DDS, as well as about the versions of the eProsima
software products related to the Fast DDS.
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: docs/notes/versions.rst [hunk 17]
```diff
@@ -0,0 +1,388 @@
...
+
+        .. list-table::
+            :header-rows: 1
+
+            * - OS \ Architecture
+              - amd64
+              - amd32
+              - arm64
+            * - Ubuntu Jammy (22.04)
+              - Tier 1: GCC 11.4 :raw-html:`<br />`
```
Comment (MiguelCompany) [inline docs/notes/versions.rst:64]: I am almost sure we have an alias for the line break

```suggestion
              - Tier 1: GCC 11.4 |br|
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
One nit apart from Eduardo's suggestions

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

File: docs/02-formalia/titlepage.rst [hunk 18]
```diff
@@ -168,6 +168,23 @@ Main Features
...
 
+Dependencies and compatibilities
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*eProsima Fast DDS* is continuously evolving and improving.
+This means that the different software products that are part of the Fast DDS ecosystem are evolving and improving
+together with Fast DDS.
+
+Fast DDS has some :ref:`library dependencies <dependencies_compatibilities_library_dependencies>`, e.g. the previously
+mentioned Fast CDR for data serialization, or OpenSSL for secure communication.
```
Comment (EduPonz) [inline docs/02-formalia/titlepage.rst:179]:
```suggestion
mentioned Fast CDR for data serialization, or OpenSSL for secure communications.
```

File: docs/notes/versions.rst [hunk 19]
```diff
@@ -0,0 +1,391 @@
...
+.. _dependencies_compatibilities:
+
+Dependencies and compatibilities
+================================
+
+Fast DDS is continuously evolving and improving.
+This means that the different software products that are part of the Fast DDS ecosystem are evolving and improving
+together with Fast DDS.
+This section provides information about the required dependencies for building Fast DDS, as well as about the versions
+of the eProsima software products related to the Fast DDS.
```
Comment (EduPonz) [inline docs/notes/versions.rst:13]:
```suggestion
of the eProsima software products related to Fast DDS.
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Review by EduPonz — APPROVED** [review]

**Pull request merged**